wire-runtime/src/test/proto-kotlin/com/squareup/wire/protos/person/Person.kt

```diff
@@ -0,0 +1,115 @@
+// Source file: person.proto
+package com.squareup.wire.protos.person
+
+import com.squareup.wire.EnumAdapter
+import com.squareup.wire.FieldEncoding
+import com.squareup.wire.ProtoAdapter
+import com.squareup.wire.ProtoReader
+import com.squareup.wire.ProtoWriter
+import com.squareup.wire.WireEnum
+import com.squareup.wire.kotlin.UnkownFieldsBuilder
+import kotlin.Int
+import kotlin.String
+import kotlin.collections.List
+import okio.ByteString
+
+data class Person(
+    val name: String,
+    val id: Int,
+    val email: String? = null,
+    val phone: List<PhoneNumber>,
+    val unknownFields: ByteString = ByteString.EMPTY
+) {
+    object ADAPTER : ProtoAdapter<Person>(FieldEncoding.LENGTH_DELIMITED, Person::class.java) {
+        override fun encodedSize(value: Person): Int = ProtoAdapter.STRING.encodedSizeWithTag(1, value.name) + 
+        ProtoAdapter.INT32.encodedSizeWithTag(2, value.id) + 
+        ProtoAdapter.STRING.encodedSizeWithTag(3, value.email) + 
+        PhoneNumber.ADAPTER.asRepeated().encodedSizeWithTag(4, value.phone) + 
+        value.unknownFields.size()
+
+        override fun encode(writer: ProtoWriter, value: Person) {
+            ProtoAdapter.STRING.encodeWithTag(writer, 1, value.name)
+            ProtoAdapter.INT32.encodeWithTag(writer, 2, value.id)
+            ProtoAdapter.STRING.encodeWithTag(writer, 3, value.email)
+            PhoneNumber.ADAPTER.asRepeated().encodeWithTag(writer, 4, value.phone)
+            writer.writeBytes(value.unknownFields)
+        }
+
+        override fun decode(reader: ProtoReader): Person {
+            var name: kotlin.String? = null
+            var id: kotlin.Int? = null
+            var email: kotlin.String? = null
+            var phone = mutableListOf<com.squareup.wire.protos.person.Person.PhoneNumber>()
+            val unknownFields = reader.decodeMessage { tag -> 
+                when(tag) {
+                    1 -> name = ProtoAdapter.STRING.decode(reader)
+                    2 -> id = ProtoAdapter.INT32.decode(reader)
+                    3 -> email = ProtoAdapter.STRING.decode(reader)
+                    4 -> phone.add(PhoneNumber.ADAPTER.decode(reader))
+                    else -> UnkownFieldsBuilder.UNKNOWN_FIELD
+                }
+            }
+            return Person(
+                name = name ?: throw com.squareup.wire.internal.Internal.missingRequiredFields(name, "name"),
+                id = id ?: throw com.squareup.wire.internal.Internal.missingRequiredFields(id, "id"),
+                email = email,
+                phone = phone,
+                unknownFields = unknownFields)
+        }
+    }
+
+    enum class PhoneType(private val value: Int) : WireEnum {
+        MOBILE(0),
+
+        HOME(1),
+
+        /**
+         * Could be phone or fax. */
+        WORK(2);
+
+        override fun getValue(): Int = value
+
+        object ADAPTER : EnumAdapter<PhoneType>(PhoneType::class.java) {
```

Comment:
You can actually combine this and the `companion object` to be
```kotlin
companion object ADAPTER : EnumAdapter<PhoneType>(PhoneType::class.java) {
  override fun fromValue(value: Int): PhoneType? = values().find { it.value == value } 
}
```

Reply:
nice catch

Reply:
After reading your comment below, I was wondering if we would still need this for java compatibility ?

Reply:
On some digging it looks like the Adapter just calls the `getValue`, i will annotate that with `@JvmStatic` and get rid of this